src/ostree/ot-main.c

```diff
@@ -379,29 +368,22 @@ ostree_admin_option_context_parse (GOptionContext *context,
     {
       /* Released when sysroot is finalized, or on process exit */
       if (!ot_admin_sysroot_lock (sysroot, error))
-        goto out;
+        return FALSE;
     }
 
   if (out_sysroot)
     *out_sysroot = g_steal_pointer (&sysroot);
 
-  success = TRUE;
-
-out:
-  return success;
+  return TRUE;
 }
 
 gboolean
 ostree_ensure_repo_writable (OstreeRepo *repo,
                              GError **error)
 {
-  gboolean ret;
-
-  ret = ostree_repo_is_writable (repo, error);
-
-  g_prefix_error (error, "Cannot write to repository: ");
```

Comment:
Heh, had to do a double take here. That's a sly piece of code. Much prefer the rewritten version.